build_rules.gradle

```diff
@@ -593,6 +593,118 @@ ext.applyJavaNature = {
   }
 }
 
+// Reads and contains all necessary performance test parameters
+class JavaPerformanceTestConfiguration {
+  String pkbLocation = System.getProperty('pkbLocation')
+
+  String logLevel = System.getProperty('logLevel', 'INFO')
+  String gradleBinary = System.getProperty('gradleBinary', './gradlew')
+  String isOfficial = System.getProperty('official', 'true')
+  String benchmarks = System.getProperty('benchmarks', 'beam_integration_benchmark')
+
+  String beamPrebuilt = System.getProperty('beamPrebuilt', 'true')
+  String beamSdk = System.getProperty('beamSdk', 'java')
+
+  String timeout = System.getProperty('itTimeout', '1200')
+
+  String kubeconfig = System.getProperty('kubeconfig', System.getProperty('user.home') + '/.kube/config')
+  String kubectl = System.getProperty('kubectl', 'kubectl')
+
+  // PerfKit will have trouble reading 'null' path. It expects empty string if no scripts are expected.
+  String kubernetesScripts = System.getProperty('kubernetesScripts', '')
+
+  String integrationTestPipelineOptions = System.getProperty('integrationTestPipelineOptions')
+
+  // PerfKit will have trouble reading 'null' path. It expects empty string if no config file is expected.
+  String optionsConfigFile = System.getProperty('beamITOptions', '')
+
+  String integrationTest = System.getProperty('integrationTest')
+  String itModule = System.getProperty('itModule')
+
+  String runner = System.getProperty('integrationTestRunner', 'direct')
+
+  String extraProperties = System.getProperty('beamExtraProperties', '')
+}
+
+// Configures a project with a set of tasks needed for running performance tests
+ext.enableJavaPerformanceTesting = {
+  println "enableJavaPerformanceTesting with ${it ? "$it" : "default configuration"} for project ${project.name}"
+
+  // Use the implicit it parameter of the closure to handle zero argument or one argument map calls.
+  JavaPerformanceTestConfiguration configuration = it ? it as JavaPerformanceTestConfiguration : new JavaPerformanceTestConfiguration()
+
+  // Add runners needed to run integration tests on
+  task packageIntegrationTests(type: Jar) {
+    if (gradle.startParameter.taskNames.contains('integrationTest')) {
```

Comment:
So, tasks names starting with 'integrationTest' is a requirement ?

Reply:
This line effectively means that we should include the runner dependencies only when the `integrationTest` task is run. So when user runs `./gradlew integrationTest...` those dependencies are added while building the jar. 

Now I see that this is not enough due the following reason: there is no way to "prebuild" tests - we cannot build with a separate build command and then just run the tests. Such prebuilding is needed by perfkit so it's a necessity I forgot to take into account here. "Prebuilding" is invoked when we specify the "beam_prebuilt" flag to false, see more in [Perfkit's code](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/a3e6c0382809cb68cd294f34cbd5b3b558e53f56/perfkitbenchmarker/beam_benchmark_helper.py#L147).

I think we should remove this condition and include the runner dependency only if the command is run with appropriate "integrationTestRunner" property . I changed this in the next commit. 
So now we can prebuild in Perfkit using this command: 
`./gradlew clean build -xtest` -DintegrationTestRunner=dataflow 
and run the tests using `integrationTest` task, for example like this: 

and execute the tests using this one:
`./gradlew clean integrationTest -p sdks/java/io/file-based-io-tests/ -DintegrationTestPipelineOptions='[]' --tests org.apache.beam.sdk.io.text.TextIOIT  -DintegrationTestRunner=dataflow`

WDYT?